Adds procedure to create private cluster with private image registry (#69087)
openshift/openshift-docs: modules/minimum-required-permissions-ipi-azure.adoc

* `Microsoft.Storage/storageAccounts/write`
====

> .Required permissions for creating a private image registry endpoint

Reviewer: since the step to configure internal image-registry is optional, here should be "Optional permission" instead of "Required permission", so that user only requires those permissions when configuring the step.

suggest to change `creating a private image registry endpoint` to `creating a image registry storage account private endpoint`, @flavianmissi  wdyt?

Reviewer 2: That does make sense @jinyunma !
It's worth mentioning that I added these permissions to the registry operator credentials request: https://github.com/openshift/cluster-image-registry-operator/pull/971/files
If we are to merge this then I think we won't need to document this.
WDYT?

> .Optional permissions for creating an image registry storage account private endpoint

Reviewer 3: Maybe  replace Optional with Required

Reviewer 2: +1
Making the storage account private is optional, but when users choose to do so then these permissions are required.

Author: @wewang58 @flavianmissi Set to "Required"

Reviewer 3: @stevsmit  Sorry, after discussed with @jinyunma  it should Optional, it means it's just a optional permission only you when you create an image registry storage account private endpoint, can you revert it again, thanks

> .Required permissions for creating an image registry storage account private endpoint

Author: @jinyunma Should the required permissions here be updated to include the following? or should we leave it as is. 

```suggestion
      - Microsoft.Storage/storageAccounts/blobServices/read
      - Microsoft.Storage/storageAccounts/blobServices/containers/read
      - Microsoft.Storage/storageAccounts/blobServices/containers/write
      - Microsoft.Storage/storageAccounts/blobServices/generateUserDelegationKey/action
      - Microsoft.Storage/storageAccounts/read
      - Microsoft.Storage/storageAccounts/write
      - Microsoft.Storage/storageAccounts/delete
      - Microsoft.Storage/storageAccounts/listKeys/action
      - Microsoft.Resources/tags/write
      - Microsoft.Network/privateEndpoints/write
      - Microsoft.Network/privateEndpoints/read
      - Microsoft.Network/privateEndpoints/privateDnsZoneGroups/write
      - Microsoft.Network/privateEndpoints/privateDnsZoneGroups/read
      - Microsoft.Network/privateDnsZones/read
      - Microsoft.Network/privateDnsZones/write
      - Microsoft.Network/privateDnsZones/join/action
      - Microsoft.Network/privateDnsZones/A/write
      - Microsoft.Network/privateDnsZones/virtualNetworkLinks/write
      - Microsoft.Network/privateDnsZones/virtualNetworkLinks/read
      - Microsoft.Network/virtualNetworks/subnets/join/action
      - Microsoft.Network/virtualNetworks/join/action
      - Microsoft.Network/networkInterfaces/read
      - Microsoft.Storage/storageAccounts/PrivateEndpointConnectionsApproval/action
  ```

Reviewer: I think we should leave it as is now, no need to update.
Because you could find those new permissions(compared with the original permissions list) have already been added in other sections (required permissions for xxxxx) in doc [installing-azure-account#minimum-required-permissions-ipi-azure_installing-azure-account](https://69087--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-account#minimum-required-permissions-ipi-azure_installing-azure-account)) , so no need to duplicate them here.

> .Required permissions for creating an image registry storage account private endpoint

Reviewer: The required permission in this doc is that it is always required when you deploying IPI cluster whatever the configuration. But here it is optional operation to create an image registry storage account private endpoints and only exist on private cluster, so suggest to change `Optional permissions`

Author: Renamed ".Optional permissions for creating an image registry storage account private endpoint"

> .Optional permissions for creating an image registry storage account private endpoint

Reviewer 2: ```suggestion
.Optional permissions for creating a private storage endpoint for the image registry
```
